vyper/semantics/types/module.py

```diff
@@ -559,3 +559,19 @@ def immutable_section_bytes(self):
     @cached_property
     def interface(self):
         return InterfaceT.from_ModuleT(self)
+
+    @cached_property
+    def is_stateless(self):
+        """
+        Determine whether ModuleT is stateless by examining its top-level
+        declarations. A module has state if it contains storage variables,
+        transient variables, or immutables, or if it includes a "uses" or
+        "initializes" declaration.
+        """
+        if len(self.initializes_decls) > 0 or len(self.uses_decls) > 0:
```

Comment:
should we really check for `uses_decl`? `uses` refer to another's module state. or maybe i'm just debating the definition of "statlessness" - if it's about not accessing any state (in the sense of purity) then this code's ok